docs/react-v9/contributing/rfcs/react-components/convergence/z-index-handling.md

```diff
@@ -0,0 +1,319 @@
+
+### What will be covered
+
+This proposal utilize this elevation system to define layers that translate into z-axis values, and those can be used by the components. The concepts are:
+
+- Each layer have a well-thought z-index number mapped to it
+- Multiple components can occupy the same layer, if they share the same importance level in the stack (e.g. Drawer and Nav)
+- A component can be defined without a z-index. Its placement fallbacks to its position within the [stack context](https://web.dev/learn/css/z-index/#stacking-context).
+
+### What won't be covered
+
+- Negative z-indexes. Components can define negative z-index values to represent background objects. Since this is very subjective, it'll be up to the component to define its own negative z-index values.
+- Correlation between shadows and z-indexes. Even though it might be related, this is out of the scope of this proposal. If necessary, it can be covered by a follow-up RFC.
+
+## Options
+
+### Issues
+
+The approaches below only standardize the layers and define z-index values for them. This is great for the current state of our components that can define arbitrary values, but won't solve a very specific problem: Defining priority for similar UI elements. e.g. Two Dropdowns are created. Which of them should have higher priority and therefore be displayed on top?
+
+#### Option A: Expose a named map of layers to z-index values (similarly to typography styles)
+
+Exposes a named map of layers to z-index values. This map can be used by components to define their z-index values.
+This would required the creation of a new file and export under `@fluentui/tokens`, to make the map available.
+
+The layers can be defined as follows:
+
+```ts
+// packages/tokens/src/global/zIndexes.ts
+
+import { ZIndexLayers } from '../types';
+
+/**
+ * Global z-index values for elements
+ */
+export const zIndexes: ZIndexLayers = {
+  background: 0, // default
+  content: 1, // content - header, footer, sidebar
+  overlay: 1000, // overlay elements - drawer, nav
+  popup: 2000, // popup layers - popups, modals, dialogs
+  messages: 3000, // communication elements - banners, messages, toasts, snackbar
+  floating: 4000, // floating elements - dropdowns, teaching
+  priority: 5000, // priority elements - tooltips
+  debug: 6000, // debug - error overlays, debug messages
+};
+```
+
+```ts
+// partner side
+
+import { zIndexes } from '@fluentui/theme';
+
+const styles = {
+  root: {
+    zIndex: zIndexes.overlay,
+  },
+};
+```
+
+<!-- FIXME: Update section after alignment with designers -->
+
+_NOTE: The names above are not final and only serve as an example. As of the moment this RFC was introduced, there are ongoing discussions with the Design team to define all the layers. This RFC will be updated before the final approval to include the definitive names._
+
+##### Pros
+
+- 👍 Clear separation by group of UI elements
+- 👍 Easy to override, extend and update
```

Comment:
I don't think that "easy to override" is true there as values are hardcoded into the object 😛

Reply:
Easy to override in our side. I will add a note to clarify that.